lib/upload_creator.rb

```diff
@@ -130,6 +130,16 @@ def create_for(user_id)
       end
 
       add_metadata!
+
+      if SiteSetting.secure_media
+        @upload.assign_attributes(
+          @upload.secure_params(
+            UploadSecurity.new(@upload, @opts.merge(creating: true)).should_be_secure_with_reason,
+            "upload creator"
+          )
+        )
```

Comment:
Looks like it's trying to do too much on the "same line" 

```suggestion
        secure, reason =  UploadSecurity.new(@upload, @opts.merge(creating: true)).should_be_secure_with_reason
        attrs = @upoad.secure_params(secure, reason, "upload creator")
        @upload.assign_attributes(attrs)
```

Or something similar.